core/drivers/wdt/watchdog_sm.c

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 Microchip
+ */
+
+#include <drivers/wdt.h>
+#include <sm/optee_smc.h>
+#include <sm/psci.h>
+
+enum smcwd_call {
+	SMCWD_INIT		= 0,
+	SMCWD_SET_TIMEOUT	= 1,
+	SMCWD_ENABLE		= 2,
+	SMCWD_PET		= 3,
+	SMCWD_GET_TIMELEFT	= 4,
+};
+
+static unsigned long wdt_min_timeout;
+static unsigned long wdt_max_timeout;
+
+enum sm_handler_ret wdt_sm_handler(struct thread_smc_args *args)
```

Comment:
No one's calling this function.

Reply:
I will add a commit adding the call to `wdt_sm_handler()`